src/Experimental/ShortestPaths/astar.jl

```diff
@@ -0,0 +1,126 @@
+# Parts of this code were taken / derived from Graphs.jl. See LICENSE for
+# licensing details.
+
+# A* shortest-path algorithm
+
+struct AStar{F<:Function} <: ShortestPathAlgorithm
```

Comment:
Maybe use `Callable` instead  of `Function` here.

Reply:
I don't know about `Callable`. What's the difference?

Reply:
Apparently it is Base.Callable. It is typeunion of `Function`  and `Type` - probably because you can also implement calls on types.

Seems that the whole thing is a bit disputed: https://github.com/JuliaLang/julia/issues/16191

Reply:
Yeah. I'll leave it as `Function` for now since `Callable` isn't exported and is a `Union` in any case. We can always change it later (non-breaking).